### PR TITLE
Add more specific SVG rule

### PR DESCRIPTION
**Pull request by egibs**
A recently-added third-party SVG rule is causing several new false-positives. The rule content only looks for two strings:
```yara
rule DELIVRTO_SUSP_SVG_Foreignobject_Nov24
{
	meta:
		description = "Presence of foreignObject in SVG file"
		author = "delivr.to"
		id = "148ef54a-4389-58c9-be89-c1714ef08371"
		date = "2024-11-28"
		modified = "2024-11-28"
		reference = "https://github.com/delivr-to/detections"
		source_url = "https://github.com/delivr-to/detections/blob/29cddf297f5eaff884b0e1c201bd9b7447022cf8/yara-rules/svg_foreignobject.yar#L1-L12"
		license_url = "N/A"
		logic_hash = "51fdc105e826344b9e516a35178c37b6e4620781ee3a9ae64b9be181a13292e7"
		score = 40
		quality = 51
		tags = ""

	strings:
		$svg = "svg" ascii wide nocase
		$fo = "</foreignObject" ascii wide nocase

	condition:
		all of them
}
```

This PR turns that rule off and adds three new SVG rules:
- `svg`
- `foreign_object` (requires `svg` to be true)
- `foreign_object_script` (requires both `svg` and `foreign_object` to be true)

The critical rule requires that an embedded SVG file use foreignObject tags along with one of three distinct patterns (images/input, base64/obfuscated variables, or xhtml/xmlns). This should cut down on the number of false positives we see.

I tested this with one of the false positive findings (`kuma-cp`) and only the SVG rule alerts now:
```
≡ data [MEDIUM]
│       🟡 base64/external — calls base64 command to decode strings: base64 -d
│       🔵 compression/bzip2 — Works with bzip2 files
│       🔵 compression/gzip — works with gzip files
│       🔵 compression/zlib — uses zlib
│       🔵 compression/zstd — Zstandard: fast real-time compression algorithm: zstd
│       🟡 embedded/base64_url — Contains base64 url: h0dHA6Ly::$http, odHRwOi8v::$http
│       🟡 embedded/html — Contains HTML content: <html lang, <html>, DOCTYPE html
│       🔵 embedded/pem_test_key — Contains TESTING KEY directive: TESTING KEY-----
│       🟡 embedded/svg — Contains SVG file(s): </svg>, <svg
│       🔵 encoding/base64 — Supports base64 encoded strings
│       🔵 encoding/json — Supports JSON encoded objects: encoding/json
│       🔵 encoding/json_decode — Decodes JSON messages: JSON.parse, json.Unmarshal
│       🔵 encoding/json_encode — encodes JSON: JSON.stringify, JSONEncode
│       🔵 hash/md5 — Uses the MD5 signature format: md5:
````

### CODE REVIEW

**Review by hectorj2f — APPROVED** [review]

File: rules/data/embedded/embedded-svg.yara [hunk 1]
```diff
@@ -0,0 +1,44 @@
+rule svg: medium {
+  meta:
+    description = "Contains SVG file(s)"
```
Comment (tstromberg) [inline rules/data/embedded/embedded-svg.yara:3]: It'd be more accurate if this said: `contains SVG (Scalable Vector Graphics) content`

File: rules/data/embedded/embedded-svg.yara [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  strings:
+    $svg_open  = "<svg>" nocase
+    $svg_close = "</svg>" nocase
```
Comment (tstromberg) [inline rules/data/embedded/embedded-svg.yara:7]: avoid `nocase` when possible - https://github.com/Neo23x0/YARA-Performance-Guidelines

Basically unless we've seen folks trying to evade detection via case sensitivity, I'd add both a caps & lowercase example.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: rules/data/embedded/embedded-svg.yara (hunk 2 above)
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Updated in [`00ddb63` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/00ddb6315da42f1d23b32da5a91293492291a916).

File: rules/data/embedded/embedded-svg.yara (hunk 1 above)
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Updated in [`00ddb63` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/00ddb6315da42f1d23b32da5a91293492291a916).

File: rules/data/embedded/embedded-svg.yara [hunk 3]
```diff
@@ -0,0 +1,44 @@
+rule svg: medium {
```
[the remark this answers is not in the document]
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Updated in [`00ddb63` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/00ddb6315da42f1d23b32da5a91293492291a916).

File: rules/data/embedded/embedded-svg.yara [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    description = "Contains SVG (Scalable Vector Graphics) content"
+
+  strings:
+    $svg_open1  = "<svg>"
+    $svg_open2  = "<SVG>"
+    $svg_close1 = "</svg>"
+    $svg_close2 = "</SVG>"
+
+  condition:
+    all of them
```
Comment (tstromberg) [inline rules/data/embedded/embedded-svg.yara:12]: you probably want something like `all of ($upper*) or all of ($lower*)`

File: rules/data/embedded/embedded-svg.yara [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+rule foreign_object: high {
```
Comment (tstromberg) [inline rules/data/embedded/embedded-svg.yara:15]: reading through https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject - I think this should be a "medium".

File: rules/data/embedded/embedded-svg.yara [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  strings:
+    $foreign_obj_open  = "<foreignObject>" nocase
+    $foreign_obj_close = "</foreignObject>" nocase
+
+  condition:
+    svg and all of them
+}
+
+rule foreign_object_script: critical {
```
Comment (tstromberg) [inline rules/data/embedded/embedded-svg.yara:27]: This feels more like a `high` to me; I think we'll find at least one false-positive for it in the future.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: rules/data/embedded/embedded-svg.yara (hunk 4 above)
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Good catch -- updated in [`efa05ad` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/efa05ad402738bbeda438125be14027e84461b5d).

File: rules/data/embedded/embedded-svg.yara (hunk 5 above)
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Updated in [`19224b3` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/19224b38d27b23d94c32d35dda2d5b3bebfe0121).

File: rules/data/embedded/embedded-svg.yara (hunk 6 above)
Comment (egibs) [reply rules/data/embedded/embedded-svg.yara]: Updated in [`19224b3` (#704)](https://github.com/chainguard-dev/malcontent/pull/704/commits/19224b38d27b23d94c32d35dda2d5b3bebfe0121).

**Review by tstromberg — APPROVED** [review]

**Pull request merged**